src/TmsIntegrationExample.Services/TmsGatewayApi/TmsGatewayClient.cs

```diff
@@ -0,0 +1,96 @@
+﻿// <copyright file="TmsGatewayClient.cs" company="Engaged Technologies">
+//      Copyright (c) Engaged Technologies. All rights reserved.
+//  </copyright>
+// <summary>
+//     Implementation of ITmsGatewayClient
+// </summary>
+
+namespace TmsIntegrationExample.Services.TmsGatewayApi
+{
+    using IdentityModel;
+    using IdentityModel.Client;
+    using Microsoft.Extensions.Options;
+    using Newtonsoft.Json;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Net.Http.Json;
+    using System.Text;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using TmsIntegrationExample.Services.Extensions;
+    using TmsIntegrationExample.Services.TmsGatewayApi.Configuration;
+    using TmsIntegrationExample.Services.TmsGatewayApi.Model;
+
+    public class TmsGatewayClient : ITmsGatewayClient
+    {
+        private readonly JsonSerializerOptions jsonSerializerOptions;
+        private readonly HttpClient httpClient;
+        private readonly TmsGatewayConfiguration tmsGatewayConfiguration;
+        private TokenResponse accessToken;
+
+        public TmsGatewayClient(
+            HttpClient httpClient,
+            IOptions<TmsGatewayConfiguration> tmsGatewayConfiguration,
+            JsonSerializerOptions jsonSerializerOptions)
+        {
+            this.tmsGatewayConfiguration = tmsGatewayConfiguration.Value;
+            this.httpClient = httpClient;
+            this.httpClient.BaseAddress = this.tmsGatewayConfiguration.BaseAddress;
+            this.jsonSerializerOptions = jsonSerializerOptions;
+        }
+
+        public async Task<Shipment> PostShipmentAsync(ShipmentPost shipment)
+        {
+            accessToken = await GetAccessTokenAsync();
```

Comment:
Do we care about linting at all? If so, this and many other references should be prefixed by `this.`.

Reply:
this prefix added into some classes.